src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -149,9 +234,27 @@ else if (getOwnBuilding() != null)
             }
         }
 
-        for (final ItemStack item : itemsNeeded)
+        
+        @Nullable final AbstractBuildingWorker building = getOwnBuilding();
+        if (building != null)
         {
-            checkIfRequestForItemExistOrCreateAsynch(item);
+            final List<GuardItemsNeeded> itemList = itemsNeeded.get(Integer.valueOf(building.getBuildingLevel()));
+            if (itemList != null)
+            {
+                final int level = worker.getCitizenData().getLevel();
+                for (final GuardItemsNeeded item : itemList)
+                {
+                	//Could have multiple armor request,  make sure the require armor falls into the guard
+                	//level.
+                	if (level >= item.getMinLevelRequired() && level <= item.getmaxLevelRequiMed())
```

Comment:
i've spotted a typo, rightmost getmaxLevelRequiMed() call should've been named `getMaxLevelRequired`, same as below at :791